fishproviz/trajectory/feeding_shape.py

```diff
@@ -89,36 +89,44 @@ def get_feeding_cords(data, camera_id, is_back):
 
 
 def get_feeding_box(data, TL_x, TL_y, TR_x, TR_y):
-    scale = 1
+    cm_len = config.MAGNET_LENGTH_CM
     # if x has the same value.
     if abs(TL_x - TR_x) < abs(TL_y - TR_y):
         # config.FRONT
-        p_len = abs(TL_y - TR_y) * scale
-        f1 = data["xpx"] > TL_x - (2 * p_len)
-        f2 = data["ypx"] < TL_y + (p_len/2)
-        f3 = data["ypx"] > TR_y - (p_len/2)
+        p_len = abs(TL_y - TR_y)
+        target_width = config.FEEDING_SHAPE_WIDTH * p_len / cm_len
+        target_height = config.FEEDING_SHAPE_HEIGHT * p_len / cm_len
+        midpoint_y = abs(TL_y - TR_y)/2 + min(TL_y, TR_y)
+        midpoint_x = TL_x
+        f1 = data["xpx"] > midpoint_x - target_height
+        f2 = data["ypx"] < midpoint_y + target_width/2
+        f3 = data["ypx"] > midpoint_y - target_width/2
         box = np.array(
             [
-                (TL_x, TL_y + (p_len / 2)),
-                (TL_x - (2 * p_len), TL_y + (p_len / 2)),
-                (TR_x - (2 * p_len), TR_y - (p_len / 2)),
-                (TR_x, TR_y - (p_len / 2)),
-                (TL_x, TL_y + (p_len / 2)),
+                (midpoint_x, midpoint_y + target_width/2),
+                (midpoint_x - target_width, midpoint_y + target_width/2),
+                (midpoint_x - target_width, midpoint_y - target_width/2),
```

Comment:
`midpoint_x - target_height` need to be replace `midpoint_x - target_width` for the x-positions